BSP/Inc/BSP.h

```diff
@@ -0,0 +1,100 @@
+#ifndef BSP_CONF_H
+#define BSP_CONF_H
+
+/**
+ * Common features / necessities used on the BSP layer of
+ * Embedded Sharepoint
+ * 
+ * Has a struct used for 
+ */
+#include <stdbool.h>
+#include "stm32f4xx_hal.h"
+#include "stm32f4xx_hal_rcc.h"
+#include "stm32f4xx_hal_gpio.h"
+#include "stm32f4xx_hal_i2c.h"
+#include "stm32f4xx_hal_spi.h"
+#include "stm32f4xx_hal_uart.h"
```

Comment:
I don't think you need these, as _hal will already do this (line 11)

Reply:
correct